common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -25,6 +25,8 @@
 import com.google.common.collect.Lists;
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.util.DiskChecker;
```

Comment:
This NM work preserving is introduced in Hadoop 2.6, and some vender release already enabled this feature by default for a long time, it is quite important for rolling upgrade (NM restart).